.github/workflows/cli-release.yml

```diff
@@ -0,0 +1,83 @@
+name: CLI Release
+
+on:
+  workflow_dispatch:
+    inputs:
+      draft-release:
+        description: 'Draft release?'
+        required: false
+        type: boolean
+        default: true
+      prerelease:
+        description: 'Prerelease?'
+        required: false
+        type: boolean
+        default: true
+      npm-version:
+        description: 'NPM version bump?'
+        required: false
+        type: choice
+        options: 
+          - major
+          - minor
+          - patch
+        default: 'patch'
+
+  # schedule:
+  #   - cron: '0 12 * * 2'  # Runs every Tuesday at 12:00 PM (noon) ET
+permissions:
+  contents: write
+
+jobs:
+  cli-release:
+    runs-on: ubuntu-latest
+    steps:      
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```

Comment:
I think I needed to add this option alongside `fetch-depth` to have permission to push to the main branch from the action:
```
token: ${{ secrets.AUTOMATION_USER_TOKEN }}
```
That secret is defined on the DevCycleHQ organization, and seems to permission to bypass the branch protection rules.

Reply:
Yeah, you might need it for both. I think it's needed anywhere you're pushing commits to a repo that has branch protection turned on.